src/main/java/kohn/maze/Maze.java

```diff
@@ -128,13 +125,38 @@ private boolean containsUnvisitedCells() {
         return false;
     }
 
-    private void displayMaze() {
+    public int getRows() {
+        return rows;
+    }
+
+    public int getCols() {
+        return cols;
+    }
+
+    public boolean getNorthCell(int x, int y) {
```

Comment:
`getNorthCell()` sounds like it's returning the North Cell itself. You want to name this (and the other methods) something like `hasNorthCell()`